### PR TITLE
Making the 'RestoreSources' configurable: sdk

### PR DESCRIPTION
**Pull request by ghost**
Remove NuGet source references - NuGet.config
Set 'RestoreSources' in RestoreSources.props - import to: Common.props
Fixed one test: Microsoft.NET.Publish.Tests.GivenThatWeWantToStoreAProjectWithDependencies.store_nativeonlyassets

Unrelated:
README.md -formatting
tools/PushToBlobFeed.targets - helpful comment

@dotnet/dotnet-cli
@jcagme
@karajas
@chcosta

### CODE REVIEW

File: Common.props [hunk 1]
```diff
@@ -55,6 +55,19 @@
...
+    <RestoreSources>
+      https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json;
+      https://www.myget.org/F/nugetbuild/api/v3/index.json;
+      https://dotnet.myget.org/F/msbuild/api/v3/index.json;
+      https://dotnet.myget.org/F/dotnet-buildtools/api/v3/index.json;
+      https://dotnet.myget.org/F/dotnet-cli/api/v3/index.json;
+      https://dotnet.myget.org/F/dotnet-core/api/v3/index.json;
+      https://dotnet.myget.org/F/roslyn-master-nightly/api/v3/index.json;
+      https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json;
+      https://api.nuget.org/v3/index.json
```
Comment (nguerrera) [inline Common.props]: Aside, but I think we can clean out a bunch of these.

File: TestAssets/TestProjects/AppWithLibraryAndRid/LibraryWithRid/LibraryWithRid.csproj [hunk 2]
```diff
@@ -2,6 +2,7 @@
   <PropertyGroup>
     <TargetFramework>netstandard1.4</TargetFramework>
     <RuntimeIdentifier>$(TestRuntimeIdentifier)</RuntimeIdentifier>
+    <RestoreSources>$(RestoreSources)</RestoreSources>
```
Comment (nguerrera) [inline TestAssets/TestProjects/AppWithLibraryAndRid/LibraryWithRid/LibraryWithRid.csproj]: This looks like a no-op. Setting ResotureSources to itself.

**Review by nguerrera — CHANGES_REQUESTED** [review]
csproj changes look like unnecessary no-ops

File: NuGet.config [hunk 3]
```diff
@@ -1,20 +0,0 @@
...
-    <add key="dotnet-core" value="https://dotnet.myget.org/F/dotnet-core/api/v3/index.json" />
-    <add key="roslyn-master-nightly" value="https://dotnet.myget.org/F/roslyn-master-nightly/api/v3/index.json" />
-    <add key="roslyn-tools" value="https://dotnet.myget.org/F/roslyn-tools/api/v3/index.json" />
-    <add key="api.nuget.org" value="https://api.nuget.org/v3/index.json" />
-  </packageSources>
-  <config>
-    <add key="repositoryPath" value="packages" />
-    <add key="globalPackagesFolder" value="packages" />
-  </config>
-</configuration>
```
Comment (nguerrera) [inline NuGet.config]: Should we still have one with <clear /> to ensure that nothing is just using the machine config?

Test assets won't import Common.props, so I suspect that's what tests are doing now.

Comment (johnbeisner) [reply NuGet.config]: Put back a NuGet.config with just the 'config' values.

Comment (nguerrera) [reply NuGet.config]: My comment got rendered wrong (edited). I think we should have a `<clear/>` here to ensure that things are actually using RestoreSources and not the machine/default config. I suspect tests are doing the latter.

**Review by livarcocc — APPROVED** [review]

**Comment by nguerrera** [comment]
My comment is now buried in out of date hunk, but it seems tests are now using global default sources, which is bad for repeatability.

**Comment by livarcocc** [comment]
@nguerrera @johnbeisner mentioned that you talked about this with him and that he was going to fix that.

File: build/build.proj [hunk 4]
```diff
@@ -105,16 +105,23 @@
...
           LogStandardErrorAsError="true"
           WorkingDirectory="$(TestsDirectory)"
           />
   </Target>
 
+  <Target Name="CopyGlobalTestNuGetConfig">
+    <PropertyGroup>
+      <NuGetConfigToCopy>$(RepositoryRootDirectory)TestAssets\NuGet.config</NuGetConfigToCopy>
+    </PropertyGroup>
+    <Copy SourceFiles="$(NuGetConfigToCopy)" DestinationFolder="$(OutputPath)Tests\" />
```
Comment (nguerrera) [inline build/build.proj]: When this merges with repo toolset, it will ned to change.

**Comment by nguerrera** [comment]
Looks good. Do you mind squash merging this since it'll be hard to follow those back and forth commits in the future.

**Review by nguerrera — APPROVED** [review]
